src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -126,6 +144,9 @@ def set_failed(self, error: Error) -> None:
 
     def extensions_launch(self) -> None:
         vm = self.get_vm()
+        if isinstance(vm, Error):
+            self.set_failed(vm)
+            return
```

Comment:
What was the reason behind these changes?

Reply:
get_vm is ether getting existing VM or creating a new one (and we don't know what is going to happen when we call the function) and since get_vm might create a new VM and therefore we might need to associate that VM with NSG and this operation might fail

Reply:
`get_vm` doesn't create a VM if it doesn't exist.  you have to call `VM.create()` to do that.